packages/cli/commands/add/preconditions.ts

```diff
@@ -38,13 +39,9 @@ export function getGlobalPreconditions(
 			{
 				name: 'supported environments',
 				run: () => {
-					const addersForInvalidEnvironment = adders.filter((a) => {
-						const supportedEnvironments = a.environments;
-						if (projectType === 'kit' && !supportedEnvironments.kit) return true;
-						if (projectType === 'svelte' && !supportedEnvironments.svelte) return true;
-
-						return false;
-					});
+					const addersForInvalidEnvironment = adders.filter(
+						(a) => !adderSetupResult[a.id].available
```

Comment:
it might be better if `available` was replaced with something like `adderSetupResult[a.id].unsupported` that holds an array of messages that we collect from all of the add-ons and then throw at the end:
```ts
const messages = adders.flatMap((a) => adderSetupResult[a.id].unsupported.map((msg) => `${a.id}: ${msg}`));

// ...

throw new UnsupportedError(messages.join('\n'));
```

Usage in the add-on would then be:
```ts
setup: ({ kit, unsupported, dependencyVersion }) => {
	if (!kit) unsupported("Requires SvelteKit");
	if (!dependencyVersion('eslint')) unsupported("Requires eslint");
	// ...
},
```

Reply:
`unsupported` is wayyy better than `unavailable` 👍 

And adding a message sound like a perfect idea.
Working on it.